packages/react-components/src/shared/src/useChainedEventCallback.ts

```diff
@@ -2,16 +2,16 @@
 
 import { useEventCallback } from "./useEventCallback";
 
-export function createChainedFunction(...funcs) {
-    return (...args) => {
-        funcs
-            .filter(Boolean)
+export function createChainedFunction<T extends (...args: any[]) => void>(...funcs: (T | false)[]) {
```

Comment:
Instead of the extends thing wouldn't it be easier to read if you introduced an interface, something like `ChainableFunction` ?